### DISCUSSION TITLE
Feature Request: add esm build output

### DISCUSSION THREAD

**Discussion by xiaoxiangmoe** [verdaccio/verdaccio#2339] [Ideas]
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

If we build verdaccio into es module, it will help us convert to Pure ESM package in the future.

We build source files in `src` dir to cjs files in `build` dir, now.

Can we build source files to esm into `build-es` dir?

If you think it is great, I'll create an PR for it.

**Comment by juanpicado** [comment]
I like the idea, just let me share some plans I have on mind and I've working on https://github.com/verdaccio/verdaccio/pull/2178

Currently as you already know, the build files are hosted on `build/` folder and in that way is shipped, my plan is eventually bundle every dependency in a one single file (you can see an experiment here https://github.com/verdaccio/verdaccio/tree/6.x/packages/standalone), so eventually next major would be a single file and also would need to have ESM support.

Even if this inflate a bit the package size, is fine with can live with that until Node v15 is the base supported version (currently is v12), I think is a good addition. Could you share if you have on mind any specific usage of verdaccio with esm modules?

**Comment by juanpicado** [comment]
Please use as base branch `https://github.com/verdaccio/verdaccio/tree/5.x/`

**Comment by stale** [comment]
Hi pal 👋🏼!
This issue has gone quiet 😶.

 We get a lot of issues, so we currently close issues after 180 days of inactivity. It’s been at least 90c days since the last update here.
 If we missed this issue or if you want to keep it open, please reply here. You can also add/suggest the label "discuss" to keep this issue open!
 As a friendly reminder: the best way to see this issue, or any other, fixed is to open a Pull Request. Check out [https://github.com/verdaccio/contributing](https://github.com/verdaccio/contributing) for more information about opening PRs, triaging issues, and contributing!

 Thanks for being a part of the Verdaccio community! 💘
